### ISSUE TITLE
Should the type of key in AsyncHBaseByteArrayStore be string instead of Array[Byte]?

### ISSUE THREAD

**Issue by yaochitc** [twitter/storehaus#272]
During the test, I found that the key type of multiput returned map will be Array[Byte] under current implementation, which means it is impossible to get any value from the map through a get method. If I wrapped the store instance with something like Store.convert and call the multiput method on wrapped instance,  l'll get an exception. Should this be changed?

**Comment by johnynek** [comment]
I don't follow. Can you link to the specific code? Is the point here that `Array[T]` has no good hashcode/equals?

If that is the issue, yes, we should work around. We have created wrapper types in the past to give a correct hashcode/equals. I think, for instance, ByteBuffer will work for this case.

Using a String limits us to keys that can be expressed must be valid strings.

**Comment by yaochitc** [comment]
Yes, that's the problem. As you said, use a wrapped class such as ByteBuffer maybe a better choice :)

**Comment by johnynek** [comment]
@yaochitc would you like to make a pull request? :)

**Comment by yaochitc** [comment]
Ok, I'll do it after a test. And what is more, I think the library should also make use of hbase'column:)
